clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4663,7 +4663,6 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
         if (AuxT.isWindowsMSVCEnvironment()) {
           CmdArgs.push_back("-D_MT");
           CmdArgs.push_back("-D_DLL");
-          CmdArgs.push_back("--dependent-lib=msvcrt");
```

Comment:
It is hard to tell, why the new code would work properly. It does not seem correct to me now.
The idea of the line that is removed here was to enforce dynamic MSVC RT with -fsycl switch for clang++ driver.

For CL mode the MSVC RT is set in addClangCLArgs() and CL allows either /MD (msvcrt) or /MDd (msvcrtd).
The line that was removed in this patch is for 'clang++' driver, which does not allow setting /MDd, and thus the only option for -fsycl is msvcrt.

Reply:
What RT is used in the patched compiler (i.e. clang++ -fsycl test.cpp)? If that is libcmt (which is by default), then there may be errors caused by using msvcrt in SYCL and libcmt in test.obj

Reply:
The default library (msvcrt) used is passed to the `link` step via `-defaultlib`:

`> clang++ -fsycl --target=x86_64-pc-windows-msvc ~/a.cpp -### 2>&1 | grep msvcrt
 "link.exe" "-out:a.exe" "-defaultlib:msvcrt" "-defaultlib:oldnames" "-defaultlib:sycl.lib" "/IGNORE:4078" "-libpath:lib/amd64" "-libpath:atlmfc/lib/amd64" "-nologo" "/tmp/a-ca1b1c.o" "/tmp/a-028002.o"`

The use of the `-defaultlib` option is used for `clang++` and the embedded default directives are used for `clang-cl`.